src/helper_functions.jl

```diff
@@ -669,3 +669,20 @@ function inner(x::AbstractVector{N}, A::AbstractMatrix{N}, y::AbstractVector{N}
               ) where {N}
     dot(x, A * y)
 end
+
+"""
+    rectify(x::AbstractVector{N}) where {N<:Real}
+
+Rectify a vector, i.e., take the element-wise maximum with zero.
+
+### Input
+
+- `x` -- vector
+
+### Output
+
+A copy of the vector where each negative entry is replaced by zero.
+"""
+function rectify(x::AbstractVector{N}) where {N<:Real}
+    return max.(x, zeros(N, length(x)))
```

Comment:
Avoid the zeros vector allocation with `return map(xi -> max(xi, 0), x)`; i tested it and it is ~2x faster with random vectors.

(If this function will be used in performance-critical code we can offer a `rectify!` function that relies on Julia's `map!(function, destination, collection...)`.)